File: backend/app/schemas/device.py
"""Device Pydantic schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class DeviceBase(BaseModel):
    """Base device schema."""

    hostname: str
    os_type: str = "unknown"
    os_version: str | None = None
    agent_version: str | None = None


class DeviceCreate(DeviceBase):
    """Schema for registering a device enrollment."""

    user_id: int | None = None


class DeviceRead(DeviceBase):
    """Schema for reading device data."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int | None
    last_seen: datetime
    is_compliant: bool


class DeviceList(BaseModel):
    """Paginated list of devices."""

    devices: list[DeviceRead]
    total: int

File: frontend/src/types/device.ts
export interface Device {
  id: number
  hostname: string
  os_type: string
  os_version: string | null
  agent_version: string | null
  user_id: number | null
  last_seen: string
  is_compliant: boolean
}

export interface DeviceList {
  devices: Device[]
  total: number
}

export interface DeviceCreate {
  hostname: string
  user_id?: number | null
  os_type?: string
  os_version?: string | null
  agent_version?: string | null
}

export interface AgentTokenCreate {
  name: string
}

export interface AgentTokenRead {
  id: number
  device_id: number
  name: string
  created_at: string
  last_used_at: string | null
  revoked_at: string | null
}

export interface AgentTokenIssued extends AgentTokenRead {
  token: string
}
